src/openrct2/world/Footpath.cpp

```diff
@@ -524,55 +524,50 @@ static void footpath_connect_corners(int32_t initialX, int32_t initialY, TileEle
         return;
 
     tileElement[0] = initialTileElement;
-    int32_t z = initialTileElement->base_height;
+    int32_t z = initialTileElement->GetBaseZ();
     for (int32_t initialDirection = 0; initialDirection < 4; initialDirection++)
     {
-        int32_t x = initialX;
-        int32_t y = initialY;
-        int32_t direction = initialDirection;
+        auto currentPos = footpathPos;
```

Comment:
The logic here has been changed. currentPos should be footpathPos + CoordsDirectionDelta[direction]